version: "3.7"

services:
  node_service:
    depends_on:
      pocketbase_service:
        condition: service_healthy
    build:
      context: .
      dockerfile: NodeService.Dockerfile
    volumes:
      - ./:/home/node/app/
    ports:
      - "4000:4000"
    command: > 
      npm run node_service

    networks:
      - some-net

  pocketbase_service:
    volumes:
      - ./:/home/node/app/
    build:
      context: .
      dockerfile: PocketbaseService.Dockerfile
    ports:
      - "8090:8090"
    working_dir: /home/node/app/pocketbase_service
    command: go run ppo serve --http=0.0.0.0:8090

    healthcheck:
      test: [ "CMD", "curl", "-f", "http://0.0.0.0:8090/_" ]
      interval: 30s
      timeout: 10s
      retries: 5

    networks:
      - some-net

  db:
    image: postgres:15-alpine
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - '5433:5432'
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - some-net

networks:
  some-net:
    driver: bridge
    external: true

volumes:
  db:
    driver: local